advertisementwebservices/src/com/metacube/advertisementwebservices/resources/AdvertisementResource.java

```diff
@@ -0,0 +1,130 @@
+package com.metacube.advertisementwebservices.resources;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.gson.Gson;
+import com.metacube.advertisementwebservices.enums.Status;
+import com.metacube.advertisementwebservices.facade.AdvertisementFacade;
+import com.metacube.advertisementwebservices.model.Advertisement;
+
+@Path("/AdvertisementResource")
+public class AdvertisementResource {
+    AdvertisementFacade advertisementFacade = new AdvertisementFacade();
+
+    @GET
+    @Path("/Advertisements")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String getAllAdvertisements(@HeaderParam("Authorization") String authorizationString) {
+        Map<Object, Object> resourceMap = new HashMap<Object, Object>();
```

Comment:
remove the usage of this map, replace this with Response class which contains status and payload field. And directly return Response class object in every rest method.